### PR TITLE
store attributes values in builtin types to avoid otel warnings

### PR DESCRIPTION
**Pull request by dineshyv**
# What does this PR do?

Serialize objects to built in types to avoid otel warnings

## Test Plan

╰─❯ llama stack run ~/.llama/distributions/llamastack-together/together-run.yaml

### CODE REVIEW

File: llama_stack/providers/utils/telemetry/trace_protocol.py [hunk 1]
```diff
@@ -19,17 +17,17 @@
 def serialize_value(value: Any) -> Any:
     """Serialize a single value into JSON-compatible format."""
     if value is None:
-        return None
+        return ""
     elif isinstance(value, (str, int, float, bool)):
         return value
+    elif hasattr(value, "_name_"):
```
Comment (ashwinb) [inline llama_stack/providers/utils/telemetry/trace_protocol.py:23]: what's this?!

File: llama_stack/providers/utils/telemetry/trace_protocol.py [hunk 2]
```diff
@@ -19,17 +17,17 @@
...
+        return value._name_
     elif isinstance(value, BaseModel):
-        return value.model_dump()
+        return value.model_dump_json()
```
Comment (ashwinb) [inline llama_stack/providers/utils/telemetry/trace_protocol.py:26]: should be `json.load(value.model_dump_json())`? I think the values can be plain dicts or lists right? not performing double json encoding should be important.

File: llama_stack/providers/utils/telemetry/trace_protocol.py (hunk 1 above)
Comment (dineshyv) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:23]: for handling enums.

File: llama_stack/providers/utils/telemetry/trace_protocol.py (hunk 2 above)
Comment (dineshyv) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:26]: model_dump_json gives a json string which works here. don't think we need another json.load.

Comment (ashwinb) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:26]: @dineshyv no but when you are querying it, you are going to need to do a json.loads() on some attribute values now without knowing when you should do that vs not?

File: llama_stack/providers/utils/telemetry/trace_protocol.py (hunk 1 above)
Comment (ashwinb) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:23]: @dineshyv you should check for enums explicitly 

```
elif instance(v, Enum): 
   return v.value
```

?

File: llama_stack/providers/utils/telemetry/trace_protocol.py (hunk 2 above)
Comment (dineshyv) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:26]: I see, i think we can do `serialize_value(value.model_dump())`, which should only give builtin types. What do you think?

Comment (dineshyv) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:26]: actually that does not work. i will do as you suggesed..

Comment (ashwinb) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:26]: @dineshyv yeah we do that json.loads(v.model_dump_json()) trick in many places to clean up all enums etc. recursively. it is expensive but we don't care right now.

Comment (dineshyv) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:26]: the warning is this:

` Invalid type dict for attribute 'output' value. Expected one of ['bool', 'str', 'bytes', 'int', 'float'] or a sequence of those types`

Looks like otel does not want dict as attribute values. it only supports lists. so nested query is not supported i think.

Comment (ashwinb) [reply llama_stack/providers/utils/telemetry/trace_protocol.py:26]: @dineshyv bah alright. let's keep it as is for now. can you handle the Enum bit please?

**Review by ashwinb — APPROVED** [review]

**Pull request merged**